.buildkite/pipeline.yml

```diff
@@ -87,6 +87,18 @@ steps:
     concurrency: 1
     concurrency_group: 'loop-device test'
 
+  - label: ":rotating_light: :exclamation: example tests on devmapper"
+    env:
+      DOCKER_IMAGE_TAG: "$BUILDKITE_BUILD_NUMBER"
+      EXTRAGOARGS: "-v -count=1"
+    artifact_paths:
+      - "examples/logs/*"
+    command:
+      - sudo lvcreate --type thin-pool --poolmetadatasize 16GiB --extents '50%FREE' -n build$BUILDKITE_BUILD_NUMBER fcci-vg
+      - make -C examples integ-test TEST_SS=devmapper TEST_POOL=build$BUILDKITE_BUILD_NUMBER
+      - sudo dmsetup remove_all
+      - sudo lvremove -f /dev/mapper/fcci--vg-build$BUILDKITE_BUILD_NUMBER
```

Comment:
Should we move these 2 commands to `cleanup` step below? Will they run if make fails ?

Reply:
Maybe  .buildkite/hooks/pre-exit?

Reply:
Should do the job too.

Reply:
BTW 'sudo dmsetup remove_all' would remove devices that are in-use by other tests. I may need to find a better alternative.

Reply:
As an option: devices are prefixed with pool name ([1](https://github.com/containerd/containerd/blob/master/snapshots/devmapper/snapshotter.go#L420)), so you can try removing all devices that belong to a pool. Something like
 `dmsetup list | grep pool-prefix- |  dmsetup remove`

Reply:
Updated the cleanup process. However .buildkite/hooks/pre-exit is running at every step which doesn't work well for this usecase, since I'd like to reuse the thin-pool while executing the steps.